Huddle01/cal.com
# Cal.com new Authetication flow, API integreation

## What does this PR do?



- We at huddle01 has revamped serval things including how users used to interact and communicate through huddle01 app. Huddle01 has a dedicated OAuth mechanism to empowers third party application like CalCom. So in order to make calcom work with new flow, we have re-written the current app. 
- Introduced Authentication page to let user confirm their identity before scheduling a call.  
- Huddle01 OAuth support that work independently.  
- New sets of API to communicate with huddle01 servers.  



## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?
- Once you install a fresh huddle01 app, you are required to confirm your identify via login to Huddle01 auth page. 
- Once you authentication process succeed, you can configure your events to use huddle01 app to schedule meetings when users make booking. 
- Now while booking a call, user should see Huddle01 app to book a call. 
- API calls should work and user should see the meeting link after he/she makes booking. 
- Test modify, cancel flow, we have added separate apis call correspond to each functionality. 
- In the booking pages, meeting details should be visible. 

- Are there environment variables that should be set?
Yes, We have new `HUDDLE01_API_TOKEN` to handle authentication process. It needs to be set. 

- What are the minimal test data to have?
Huddle01 account to test authenticate. 
- What is expected (happy path) to have (input and output)?
- Install huddle01 app, complete authentication process.
- Configure your events to use huddle01 app for scheduling meetings. 
- Users should be able to schedule, modify and cancel the meeting. 

## Checklist



- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
I haven't commented my code, particularly in hard-to-understand areas
I haven't checked if my changes generate no new warnings
